Added GitHub Actions

daminijain23 opened the pull request:
Added .github/workflows/tox-test.yml and .github/workflows/release.yml.

Updated tox.ini for static - 
1. constrained pylint and astroid version for travis.
2. pylint was running on src, instead it is made to run on pushcollector.

rohanpm reviewed (commented):
The content seems OK, could you please rebase on latest master and squash some of the commits together?

rajulkumar reviewed (commented):
+1 to Rohan's comment.
You may update tox.ini accordingly in this or another PR.

rajulkumar reviewed (commented):
Existing code showing up as a change in this PR for files like test-requirements.txt, tox.ini and tests/proxy/test_argument_handling.py doesn't look right. There might be some inconstancy with commit hash. Probably a rebase on latest master should fix.
@rohanpm your thoughts please.

rohanpm commented:
That's right, it would be best to do a fresh rebase here.

If we have a look at the earliest commit in this series at time of writing: https://github.com/daminijain23/pushcollector/commit/5f6dc86f82b7e23ac9cd45884e970a2a37523b60

Note the "parent" of the commit is https://github.com/daminijain23/pushcollector/commit/3b9e56d60baa32775a78c623363375d33682b21c from July 2019

This means your changes are based on top of 2+ years old work for some reason.

If rebase is done correctly, then the parent of the earliest commit in your PR should be the most recent commit from master, which at time of writing is b9e40306e196599865c680ce5e7cc16ec4b1b39e (Sept 2021).

daminijain23 committed:
Migrate to GH actions

daminijain23 commented:
Did a fresh rebase. Please review and suggest if any changes required.

rohanpm commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches:
+    - '*'
```
Why do we have branches `*` here?

I don't think it should be here.  For example, sometimes people do push development branches onto the main repo because they're too lazy to create a fork, and this should not trigger a pypi release.  I don't think there should be any `branches` here at all.

daminijain23 replied:
Corrected.

daminijain23 committed:
Update release.yml

release on pypi will not be triggered if commits are made to branches other than master.

rohanpm commented on tox.ini:
```diff
@@ -18,10 +18,11 @@ deps=
 commands=
 	black --check .
-	sh -c 'pylint src; test $(( $? & (1|2|4|32) )) = 0'
+	sh -c 'pylint pushcollector; test $(( $? & (1|2|4|32) )) = 0'
```
Hmm, does this work actually?  Wasn't "src" correct since that's where the source code is located?

daminijain23 replied:
Yes this works. Previously when it was src python was giving import errors and got resolved when code directory was updated to pushcollector here.

rohanpm replied:
Thanks.
